etcd-client/test_cmd_client.py

```diff
@@ -0,0 +1,422 @@
+import pytest
+import sys
+from os.path import abspath, dirname, join
+from subprocess import Popen
+from subprocess import check_output
+from subprocess import CalledProcessError
+
+cmd_client_path = abspath(join(dirname(__file__), "cmd_etcd_client.py"))
+
+
+def cleanup(key):
+    Popen(["etcdctl", "rm", key], stdout=None, stderr=None)
```

Comment:
Have a look at [`subprocess.run`](https://docs.python.org/3/library/subprocess.html#subprocess.run) that will do a lot of magic for you. Using `Popen` directly can sometime have issues.